### ISSUE TITLE
`leading: true` does not work in StrictMode

### ISSUE THREAD

**Issue by nklhtv** [xnimorz/use-debounce#183]
**Describe the bug**
set leading to true causes issues in StrictMode
if you type only 1 character in the example bellow the debounced value is never updated.
if you type 2 characters in less than 1 second the debounced values is updated thogh.
**To Reproduce**
```javascript
import React, { useState, StrictMode } from "react";
import ReactDOM from "react-dom";
import { useDebounce } from "use-debounce";

function Input() {
  const [text, setText] = useState("Hello");
  const [debouncedText] = useDebounce(text, 1000, { leading: true });
  return (
    <div>
      <input
        defaultValue={"Hello"}
        onChange={(e) => {
          setText(e.target.value);
        }}
      />
      <p>Actual value: {text}</p>
      <p>Debounced value: {debouncedText}</p>
    </div>
  );
}

const rootElement = document.getElementById("root");
ReactDOM.render(
  <StrictMode>
    <Input />
  </StrictMode>,
  rootElement
);
```

**Expected behavior**
To work in the same way inside and outside of StrictMode

**use-debounce version:**
10.0.3

**Comment by xnimorz** [comment]
Hey @nklhtv ,

Thanks for submitting the request. Could you check, if 10.0.1 has the same issue? My PC is in repair, and I manage to land fix after I get it back

**Comment by nklhtv** [comment]
Yes, 10.0.1 has the same issue, only in StrictMode thogh

**Comment by xnimorz** [comment]
I've published use-debounce@10.0.4-beta version, that should fix the issue. 

Please, give it a go, and let me know, if it helps. 

if so, I'll make that version stable

**Comment by xnimorz** [comment]
Published as 10.0,4. 

Feel free to re-open the issue if needed
